sixpack.js

```diff
@@ -15,16 +21,31 @@
 
     sixpack.generate_client_id = function () {
         // from http://stackoverflow.com/questions/105034
-        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+        var client_id = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
             var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
             return v.toString(16);
         });
+        if (!on_node && this.persist) {
+            document.cookie = "sixpack_client_id=" + client_id + "; expires=Tue, 19 Jan 2038 03:14:07 GMT; path=/";
+        }
+        return client_id;
     };
 
+    sixpack.persisted_client_id = function() {
+        // http://stackoverflow.com/questions/5639346/shortest-function-for-reading-a-cookie-in-javascript
+        var result;
+        return (result = new RegExp('(?:^|; )' + encodeURIComponent('sixpack_client_id') + '=([^;]*)').exec(document.cookie)) ? (result[1]) : null;
+    }
+
     sixpack.Session = function (options) {
         Object.assign(this, sixpack, options);
+
         if (!this.client_id) {
-            this.client_id = this.generate_client_id();
+            if (this.persist && !on_node) {
+                this.client_id = this.persisted_client_id() || this.generate_client_id();
```

Comment:
Hmm, you mean change this:
```
this.client_id = this.persisted_client_id() || this.generate_client_id();
```
to something like:
```
this.client_id = this.persisted_client_id() !== null ? this.persisted_client_id() : this.generate_client_id();
```
or
```
var persisted_id = this.persisted_client_id();
if (persisted_id !== null) {
    this.client_id =  persisted_id;
else {
    this.client_id = this.generate_client_id();
}
```

Or did you point to some other row? If not I think I'd still opt for the original.